lib/assert.js

```diff
@@ -300,13 +300,15 @@ function setEquiv(a, b, strict, actualVisitedObjects) {
 
   outer: for (const val1 of a) {
     if (!b.has(val1)) {
-      // The value doesn't exist in the second set by reference, so we'll go
-      // hunting for something thats deep-equal to it. Note that this is O(n^2)
-      // complexity, and will get slower if large, very similar sets / maps are
-      // nested inside. Unfortunately there's no real way around this.
-      for (const val2 of b) {
-        if (_deepEqual(val1, val2, strict, actualVisitedObjects)) {
-          continue outer;
+      if (!strict || typeof val1 === 'object') {
```

Comment:
Its a minor point, but do you think its still worth putting in the null check? I'm indifferent either way, but happy to add it if you think the marginal speed improvement is worth the marginal complexity.